open-models/serving/cloud_run_ollama_qwen3_inference.ipynb

```diff
@@ -0,0 +1,720 @@
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "tvgnzT1CKxrO"
+      },
+      "source": [
+        "> [**Qwen 3**](https://qwenlm.github.io/blog/qwen3/) is the latest generation of large language models in Qwen series, offering a comprehensive suite of dense and mixture-of-experts (MoE) models.\n",
+        "It supports thinking and function calling.\n",
+        "\n",
+        "> **[Cloud Run](https://cloud.google.com/run)**:\n",
+        "It's a serverless platform by Google Cloud for running containerized applications. It automatically scales and manages infrastructure, supporting various programming languages. Cloud Run now offers GPU acceleration for AI/ML workloads. With 30 seconds to the first token, Cloud Run is a perfect platform for serving lightweight models.\n",
+        "\n",
+        "> **Note:** [GPU support in Cloud Run](https://cloud.google.com/run/docs/configuring/services/gpu) is Generally Available.\n",
+        "To use the GPU feature, you must request `Total Nvidia L4 GPU allocation without zonal redundancy, per project per region` quota under Cloud Run in the [Quotas and system limits page](https://cloud.google.com/run/quotas#increase).\n",
+        "\n",
+        "> **[Ollama](ollama.com)**: is an open-source tool for easily running and deploying large language models locally. It offers simple management and usage of LLMs on personal computers or servers.\n",
+        "\n",
+        "This notebook showcase how to deploy [Qwen 3](https://developers.googleblog.com/en/introducing-gemma3) in Cloud Run,\n",
+        "with the objective to an API for running AI Agents built with Google [Agent Development Kit](https://google.github.io/adk-docs/).\n",
+        "\n",
+        "By the end of this notebook, you will learn how to:\n",
+        "\n",
+        "1. Deploy Qwen 3 as an OpenAI-compatible API on Cloud Run using Ollama.\n",
+        "2. Build a custom container with Ollama to deploy any Large Language Model (LLM) of your choice.\n",
+        "3. Make requests to an API hosted on Cloud Run.\n",
+        "4. Create and run an Agent that uses Qwen 3.\n",
+        "\n",
+        "We will build an agent using [Agent Development Kit](https://google.github.io/adk-docs/) - a flexible and modular model-agnostic framework for developing and deploying AI agents."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "61RBz8LLbxCR"
+      },
+      "source": [
+        "## Get started"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "aOiPjM5DEPhK"
+      },
+      "source": [
+        "### Install Google Cloud SDK\n",
+        "\n",
+        "Make sure you Google Cloud SDK is installed (try running `gcloud version`) or [install it](https://cloud.google.com/sdk/docs/install) before executing this notebook.\n",
+        "\n",
+        "> If you are running in Colab or Vertex AI workbench, you have Google Cloud SDK installed."
```

Comment:
Can you add a Python cell to install the relevant SDKs? Similar to the https://github.com/GoogleCloudPlatform/generative-ai/blob/main/notebook_template.ipynb

Reply:
Done.